janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -121,8 +121,7 @@
     public LuceneIndex(Configuration config) {
         String dir = config.get(GraphDatabaseConfiguration.INDEX_DIRECTORY);
         File directory = new File(dir);
-        if (!directory.exists()) directory.mkdirs();
-        if (!directory.exists() || !directory.isDirectory() || !directory.canWrite())
+        if ((!directory.exists() && !directory.mkdirs()) || !directory.isDirectory() || !directory.canWrite())
```

Comment:
Since you're here, would you mind enclosing the `if` statement's single-line body in `{ ... }` so that it's clear that it refers to a single statement, please? We've had other situations in this code base where indentation was counter to the actual syntax; would be nice to avoid that going forward.

Incidentally, I like how Go enforces braces for `if` statements, even for a single line, to avoid this ambiguity.